testsrc/org/mozilla/javascript/tests/Issue1206Test.java

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.javascript.tests;
+
+import org.junit.Test;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.Scriptable;
+
+/**
+ * Tests the ConsString class to ensure it properly supports String, StringBuffer and StringBuilder.
+ */
+public class Issue1206Test {
```

Comment:
I think all the tests currently just check whether the code doesn't throw an exception

Think it would be better if besides that, they'd also assert the result of cx.evaluateString  to be as expected

Reply:
Great idea!  Those tests have been updated to assert the results of those cx.evaluateString calls.

The fork/branch has been updated.  I think this might be ready for review by the Rhino team!  Thanks!